indico/migrations/versions/20200204_1201_4d263fa78830_add_category_roles_table.py

```diff
@@ -0,0 +1,46 @@
+"""Add category roles table
+
+Revision ID: 4d263fa78830
+Revises: 6a185fdcd4ee
+Create Date: 2020-02-04 12:01:02.554724
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '4d263fa78830'
+down_revision = '6a185fdcd4ee'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        'roles',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('category_id', sa.Integer(), nullable=False, index=True),
+        sa.Column('name', sa.String(), nullable=False),
+        sa.Column('code', sa.String(), nullable=False),
+        sa.Column('color', sa.String(), nullable=False),
+        sa.CheckConstraint('code = upper(code)', name='uppercase_code'),
+        sa.ForeignKeyConstraint(['category_id'], ['categories.categories.id']),
+        sa.PrimaryKeyConstraint('id'),
+        schema='categories'
+    )
+    op.create_index(None, 'roles', ['category_id', 'code'], unique=True, schema='categories')
```

Comment:
you can add this to the create_table call above instead:

    sa.Index(None, 'category_id', 'code', unique=True),

Reply:
Tested with `downgrade` and `upgrade`